components/image-builder-mk3/pkg/orchestrator/orchestrator.go

```diff
@@ -228,13 +228,50 @@ func (o *Orchestrator) Build(req *protocol.BuildRequest, resp protocol.ImageBuil
 
 	// resolve build request authentication
 	reqauth := o.AuthResolver.ResolveRequestAuth(req.Auth)
+	if req.BaseImageNameResolved != "" && !req.GetForceRebuild() {
+		wsrefstr, err := o.getWorkspaceImageRef(ctx, req.BaseImageNameResolved)
+		if err != nil {
+			return status.Errorf(codes.Internal, "cannot produce workspace image ref: %q", err)
+		}
+		wsrefAuth, err := auth.AllowedAuthForAll().GetAuthFor(ctx, o.Auth, wsrefstr)
+		if err != nil {
+			return status.Errorf(codes.Internal, "cannot get workspace image authentication: %q", err)
+		}
+
+		// check if needs build -> early return
+		exists, err := o.checkImageExists(ctx, wsrefstr, wsrefAuth)
+		if err != nil {
+			return status.Errorf(codes.Internal, "cannot check if image is already built: %q", err)
+		}
+		if exists {
+			err = resp.Send(&protocol.BuildResponse{
+				Status:  protocol.BuildStatus_done_success,
+				Ref:     wsrefstr,
+				BaseRef: req.BaseImageNameResolved,
+			})
+			if err != nil {
+				return err
```

Comment:
What error code will this error be returned with? Would it be better to wrap and return with `status.Errorf`?